storage/google/cloud/storage/bucket.py

```diff
@@ -292,12 +298,32 @@ def __init__(
         bucket,
         bucket_policy_only_enabled=False,
         bucket_policy_only_locked_time=None,
+        uniform_bucket_level_access_enabled=False,
+        uniform_bucket_level_access_locked_time=None,
```

Comment:
Ordering should match the order documented above.  The constructor probably needs to check for / resolve conflicts between `uniform_bucket_level_access_enabled` / `bucket_policy_only_enabled` and `uniform_bucket_level_access_locked_time` / `bucket_policy_only_locked_time` first, which would simplify the code setting up `data`.